src/index.js

```diff
@@ -100,24 +134,60 @@ module.exports = {
       type: 'covariates',
     }],
   }, {
+    // step two: receive the globalMeanY and averageBetaVector, then calculate sseLocal, sstLocal and varXLocal
     type: 'function',
     fn(opts) {
-      const previousData = opts.previousData;
+      const globalMeanY = opts.remoteResult.data.globalMeanY;
+      const averageBetaVector = opts.remoteResult.data.averageBetaVector;
+      const biasedX = opts.previousData.biasedX;
+      const y = opts.previousData.y;
+      const rSquared = opts.previousData.rSquared;
+      const tValue = opts.previousData.tValue;
+      const pValue = opts.previousData.pValue;
+      const betaVector = opts.previousData.betaVector;
+      const localCount = y.length;
 
-      for (let i = 0; i < opts.previousData.X.length; i += 1) {
-        previousData.X[i].splice(0, 0, 1);
-      }
+      //calculate the local r squred and t value for averageBetaVector)
+      const rSquaredLocal = regression.rSquared(biasedX, y, averageBetaVector);
+      const tValueLocal = regression.tValue(biasedX, y, averageBetaVector);
+      var tdist=distributions.Studentt(localCount-1);
+      const tcdf = tValueLocal.map(r => tdist.cdf(r));
+      const pValueLocal = n.mul(2,n.sub(1,tcdf));
 
-      const betaVector = regression.oneShot(previousData.X, previousData.y);
+      // calculate sseLocal and sstLocal
+      const sseLocal=n.sum(n.pow(n.sub(y, n.dot(biasedX, averageBetaVector)), 2));
+      const sstLocal=n.sum(n.pow(n.sub(y, n.rep(n.dim(y), globalMeanY)), 2));
+
+      // calculate varXLocal
+      const varXLocalMatrix=n.dot(n.transpose(biasedX),biasedX);
```

Comment:
Because varXLocalMatrix represents covariance matrix of local biasedX, and varXLocal represents the variance term of biasedX.